tests/components/google_wifi/test_sensor.py

```diff
@@ -34,185 +34,204 @@
-        resource = f"http://{google_wifi.DEFAULT_HOST}{google_wifi.ENDPOINT}"
-        mock_req.get(resource, status_code=200)
-        assert setup_component(
-            self.hass,
-            "sensor",
-            {"sensor": {"platform": "google_wifi", "monitored_conditions": ["uptime"]}},
-        )
-        assert_setup_component(1, "sensor")
-
-    @requests_mock.Mocker()
-    def test_setup_get(self, mock_req):
-        """Test setup with full configuration."""
-        resource = f"http://localhost{google_wifi.ENDPOINT}"
-        mock_req.get(resource, status_code=200)
-        assert setup_component(
-            self.hass,
-            "sensor",
-            {
-                "sensor": {
-                    "platform": "google_wifi",
-                    "host": "localhost",
-                    "name": "Test Wifi",
-                    "monitored_conditions": [
-                        "current_version",
-                        "new_version",
-                        "uptime",
-                        "last_restart",
-                        "local_ip",
-                        "status",
-                    ],
-                }
-            },
-        )
-        assert_setup_component(6, "sensor")
-
-
-class TestGoogleWifiSensor(unittest.TestCase):
-    """Tests for Google Wifi sensor platform."""
-
-    def setUp(self):
-        """Set up things to be run when tests are started."""
-        self.hass = get_test_home_assistant()
-        with requests_mock.Mocker() as mock_req:
-            self.setup_api(MOCK_DATA, mock_req)
-        self.addCleanup(self.hass.stop)
-
-    def setup_api(self, data, mock_req):
-        """Set up API with fake data."""
-        resource = f"http://localhost{google_wifi.ENDPOINT}"
-        now = datetime(1970, month=1, day=1)
-        with patch("homeassistant.util.dt.now", return_value=now):
-            mock_req.get(resource, text=data, status_code=200)
-            conditions = google_wifi.MONITORED_CONDITIONS.keys()
-            self.api = google_wifi.GoogleWifiAPI("localhost", conditions)
-        self.name = NAME
-        self.sensor_dict = {}
-        for condition, cond_list in google_wifi.MONITORED_CONDITIONS.items():
-            sensor = google_wifi.GoogleWifiSensor(self.api, self.name, condition)
-            name = f"{self.name}_{condition}"
-            units = cond_list[1]
-            icon = cond_list[2]
-            self.sensor_dict[condition] = {
-                "sensor": sensor,
-                "name": name,
-                "units": units,
-                "icon": icon,
+"""Tests for setting up the Google Wifi sensor platform."""
+
+
+@pytest.fixture
+def cleanUp(self):
+    """Fixture for cleanup."""
+    return self.addCleanup(self.hass.stop)
+
+
+async def test_setup_minimum(hass):
+    """Test setup with minimum configuration."""
+    resource = f"http://{google_wifi.DEFAULT_HOST}{google_wifi.ENDPOINT}"
+    requests_mock.Mocker().get(resource, status_code=200)
```

Comment:
This doesn't work as it's a context manager. The mocker will now never be applied.

Instead add `requests_mock` as a fixture to `test_setup_minimum` like this:

```python
async def test_setup_minimum(hass, requests_mock):
```

And then change this line to
```suggestion
    requests_mock.get(resource, status_code=200)
```

Reply:
Do this for all other spots.